### PR TITLE
Add robust sign up/in/out

### PR DESCRIPTION
**Pull request by gekola**

### CODE REVIEW

**Comment by gekola** [comment]
[Add authentication](https://trello.com/c/sO9aHooq/8-add-authentication)

File: config/config.exs [hunk 1]
```diff
@@ -22,6 +22,14 @@ config :logger, :console,
   format: "$time $metadata[$level] $message\n",
   metadata: [:request_id]
 
+config :guardian, Guardian,
+  allowed_algos: ["HS512"],
+  issuer: "Poscaster",
+  ttl: { 90, :days },
+  verify_issuer: true,
+  secret_key: System.get_env("GUARDIAN_SECRET_KEY") || "",
```
Comment (azhi) [inline config/config.exs]: That will be substituded on compile-time, meaning that we need to set this env variable during compilation inside docker (and on localhost), which is not so convenient. I suggest we use runtime `fn ->` with `System.get_env` there.

File: test/models/invitation_test.exs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+defmodule Poscaster.InvitationTest do
+  use Poscaster.ModelCase
```
Comment (azhi) [inline test/models/invitation_test.exs]: `Poscaster.ModelCase` [checkouts](https://github.com/poscaster/backend/blob/99ebb9417ad04ee2925251388256b66b0816278a/test/support/model_case.ex#L29) connection to DB, which is not used in these tests, so i suggest we don't use it here (alternatives are vanilla `ExUnit.case` or our own case template).

Also, it is a good practise to make tests that does not depend on global state [async](http://elixir-lang.org/docs/stable/ex_unit/ExUnit.Case.html) - that way large test suite is much faster. `CaseTemplates` can also receive this option just like `Case` do, something like `use Poscaster.ModelCase, async: true`.

File: test/models/user_test.exs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+defmodule Poscaster.UserTest do
+  use Poscaster.ModelCase
```
Comment (azhi) [inline test/models/user_test.exs]: Same here.

File: web/models/user.ex [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  def find_and_confirm_password(%{"user" => %{"login" => login, "password" => password}}) do
+    user = find_by_email_or_login(login)
+    if user do
+      if Comeonin.Bcrypt.checkpw(password, user.password_hash) do
+        {:ok, user}
+      else
+        {:error}
+      end
+    else
+      # No timing attach shall pass
```
Comment (azhi) [inline web/models/user.ex]: attach ftw!

File: web/models/user.ex [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      Comeonin.Bcrypt.dummy_checkpw()
+      {:error}
+    end
+  end
+
+  defp find_by_email_or_login(str) do
+    Poscaster.User
+    |> where([u], u.login == ^str or u.email == ^str)
+    |> limit(1)
```
Comment (azhi) [inline web/models/user.ex]: [`first/2`](https://hexdocs.pm/ecto/Ecto.Query.html#first/2) is also good.

**Pull request merged**